src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random RANDOM = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int indexFigure;
+        indexFigure = RANDOM.nextInt(4);
+        switch (indexFigure) {
+            case 0:
+                return new Circle(colorSupplier.getRandomColor(), getRandomInt());
```

Comment:
It is better to move to private method
```suggestion
                return prepareCircle();
```